### PR TITLE
afw - Resource graph query fixes

### PR DESCRIPTION
**Pull request by ReneHezser**
# Overview/Summary

Fixed the resource graph query

## Related Issues/Work Items

[AB#32807](https://dev.azure.com/CSUSolEng/12ba7e46-a92a-47f6-ad6a-fba0479234a7/_workitems/edit/32807)
[AB#32808](https://dev.azure.com/CSUSolEng/12ba7e46-a92a-47f6-ad6a-fba0479234a7/_workitems/edit/32808)
[AB#32809](https://dev.azure.com/CSUSolEng/12ba7e46-a92a-47f6-ad6a-fba0479234a7/_workitems/edit/32809)

## This PR fixes/adds/changes/removes

1. add resource graph query

### Breaking Changes

none

## As part of this Pull Request I have

- [x] Read the [Contribution Guide](https://azure.github.io/Azure-Proactive-Resiliency-Library/contributing) and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Proactive-Resiliency-Library/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Proactive-Resiliency-Library/issues) or ADO Work Items (Internal Only)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Proactive-Resiliency-Library/tree/main)
- [x] Ensured PR tests are passing
- [x] Performed testing and provided evidence (e.g. screenshot of output) for any changes associated to ARG queries and/or scripts
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

**Comment by ReneHezser** [comment]
**afw-1**
image

File: docs/content/services/networking/firewall/code/afw-1/afw-1.kql [hunk 1]
```diff
@@ -1 +1,5 @@
-// under-development
+// Azure Resource Graph Query
+// List all Azure Firewalls that are not configured with multiple availability zones or deployed without a zone
+resources
+| where type == 'microsoft.network/azurefirewalls' and (isnull(zones) or zones !contains(','))
```
Comment (ejhenry) [inline docs/content/services/networking/firewall/code/afw-1/afw-1.kql:4]:
```suggestion
| where type == 'microsoft.network/azurefirewalls'
| where array_length(zones) <= 1 or isnull(zones)
```

**Comment by ejhenry** [comment]
@Azure/aprl-networking do you want to review this networking PR before @oZakari and I review?

**Comment by ehaslett** [comment]
@ReneHezser  I am still seeing blank entries for afw-3.kql. I wonder if other alerts not related to AFW are getting pulled in unintentionally:

image

**Review by ehaslett — CHANGES_REQUESTED** [review]
@ReneHezser  I am still seeing blank entries for afw-3.kql. I wonder if other alerts not related to AFW are getting pulled in unintentionally:
image

File: docs/content/services/networking/expressroute-gateway/code/ergw-5/ergw-5.kql [hunk 2]
```diff
@@ -1 +1,16 @@
...
+| join kind=fullouter (
+    resources
+    | where type == "microsoft.network/ddosprotectionplans"
+    | mv-expand properties.virtualNetworks
+    | extend vNet = tostring(properties_virtualNetworks.id)
+    | project ddosProtectionPlan = id, vNet
+    )
+    on $left.vNet == $right.vNet
+| where ddosProtectionPlan == ''
+| project recommendationId = "afw-5", name, id = firewallId, param1 = "ddosProtectionPlan:false"
```
Comment (oZakari) [inline docs/content/services/networking/expressroute-gateway/code/ergw-5/ergw-5.kql:16]:
```suggestion
// under-development
```

Comment (oZakari) [inline docs/content/services/networking/expressroute-gateway/code/ergw-5/ergw-5.kql:16]:
```suggestion
// Azure Resource Graph Query
// List all in-scope Azure Firewall resources, where the VNet is not associated to a DDoS Protection Plan
resources
| where type == "microsoft.network/azurefirewalls"
| mv-expand properties.ipConfigurations
| project name, firewallId = id, vNet = substring(properties_ipConfigurations.properties.subnet.id, 0, indexof(properties_ipConfigurations.properties.subnet, "/subnet") - 7)
| join kind=fullouter (
    resources
    | where type == "microsoft.network/ddosprotectionplans"
    | mv-expand properties.virtualNetworks
    | extend vNet = tostring(properties_virtualNetworks.id)
    | project ddosProtectionPlan = id, vNet
    )
    on $left.vNet == $right.vNet
| where ddosProtectionPlan == ''
| project recommendationId = "afw-5", name, id = firewallId, param1 = "ddosProtectionPlan:false"
```

**Review by ehaslett — APPROVED** [review]
LGTM - thank you @oZakari

**Pull request merged**